### PR TITLE
Added Declaring Variables

### PR DESCRIPTION
**Pull request by Jon2211**
Added to beginner section: declaring an untyped, a typed and a dynamic variable

### CODE REVIEW

File: assets/content/cookbook/Beginner/declare-variables.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+[tags]: / "class"
+
+# Declare variables
+
+Create a variable without type. Although the type is not explicity declared, you cannot reassign the variable later to a different type.
```
Comment (Gama11) [inline assets/content/cookbook/Beginner/declare-variables.md]: It's not really a "variable without a type", just without a type _hint_.

**Comment by dstrekelj** [comment]
I feel this entry should also mention that Haxe allows declaration of multiple variables with the same identifier (or _redecleration_ of existing variables). I'd like to see it paired up with a short example showing the generated output, but that's just a preference.

This "issue" was raised on IRC not long ago, and it seems like a thing that might trip up newcomers. Having it in the cookbook would be smart.

File: assets/content/cookbook/Beginner/declare-variables.md (hunk 1 above)
Comment (Jon2211) [inline assets/content/cookbook/Beginner/declare-variables.md]: ok - I'll update

Comment (Jon2211) [inline assets/content/cookbook/Beginner/declare-variables.md]: or maybe without a declared type

**Comment by Jon2211** [comment]
1. Should this entry discuss base types or should that be another entry?
2.  Should this entry cover private/public/inline/static etc
3. access modifiers?

**Comment by dstrekelj** [comment]
1. I'd save base types for their own entry. There's only three of them, but they'll take up about as much space as this entry does now. Besides, covering base types on their own would allow you to go into greater detail.

2. Same as above. They are good candidates for a standalone entry in the cookbook. Perhaps even more so than base types.

3. Do you mean mutators and accessors? If yes, then I say cover them with the _access modifiers_ from the previous point. If no, then I'm not sure what you mean.

**Comment by dstrekelj** [comment]
Oh, and [here](https://groups.google.com/forum/#!topic/haxelang/Lqr7eVZEZgQ)'s another reason to cover variable redeclaration. Kind of funny this happened not long after I mentioned it here. "Speak of the devil" and all that.

**Comment by markknol** [comment]
I think we should be carefu lwhen speaking about dynamic. I suggest to leave it out here. Haxe has a great type system, dynamic is a gateway to error phrone code. I rather avoid it or have a snippets/tutorial that explains why and how to avoid it.

**Comment by Jon2211** [comment]
Made some modifications to the entry.  Removed the dynamic type entry, added example of the null state of variables when created, and made an example of redeclaring variables in the local scope vs class scope.

File: assets/content/cookbook/Beginner/declare-variables.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+Create a variable with type.
+
+```haxe
+var foo:Int = 2;
+var bar:String = "2";
+```
+
+### Inital values of variables
+
+In Haxe, variables are null until used - including number types.
```
Comment (Simn) [inline assets/content/cookbook/Beginner/declare-variables.md]: That is not true for static targets.

**Comment by Simn** [comment]
This example conflates local variables and type inference in a way which I think causes more confusion than anything. Thank you for the contribution, but I prefer not to include this example for the time being.

**Pull request closed without merging**